### PR TITLE
Approving verification requests creates new hosting provider objects

### PR DESCRIPTION
**Pull request by tortila**
Closes https://github.com/thegreenwebfoundation/admin-portal/issues/313.

## Scope of changes:
- `ProviderRequest` admin has a custom "Approve" action, that can be triggered either from the change list view (for multiple requests) or from the change view (for a single request)
Screenshot from 2023-02-17 11-31-34
Custom actions are supported by Django on the change list view, but not on the change (detail) view. The action on the change view was added by introducing a `ActionInChangeFormMixin` and extending the change view form for the `ProviderRequest` admin, as seen in [this helpful StackOverflow answer](https://stackoverflow.com/a/20379270).

- The logic of approving the request handles possible failure modes by raising a `ValueError`, and the implementation is heavily commented with the reasons for temporary limitations (e.g. related to locations and user permissions)
- `ProviderRequest` admin has now correct permissions: users within the `admin` group should be able to use the change view with read-write permissions.
- Changed the available and default options for `ProviderRequestStatus`: the request can now be `APPROVED` instead of `ACCEPTED`, and the value upon submitting the request is set to `PENDING_REVIEW`.
- Added missing fields: `Hostingprovider` now has a `description`, and related evidence documents now have a `type`.

### Non-functional changes:
- [`freezegun`](https://github.com/spulec/freezegun) added as a dev dependency - it's very useful for testing time-related values in a predictable way.
- A lot of tests to cover possible cases of approving the verification requests as well as accessing the admin of `ProviderRequest`

## To do after merging
This pull request contains new migrations that need to be executed after deployment (triggered automatically upon merging) is finished.

### CODE REVIEW

**Comment by tortila** [comment]
[Verification request: Provider-related resources are created automatically upon accepting their submission](https://trello.com/c/zf3PtZHu/5-verification-request-provider-related-resources-are-created-automatically-upon-accepting-their-submission)

**Comment by mrchrisadams** [comment]
Thanks for working on this Oliwia! I've reviewed it and I'm very happy with it, but I hope you'll understand if I don't merge this in at 17:30 on a Friday afternoon 😅 .

I intend to merge this in on Monday morning when I have fresh eyes, but broadly speaking there's nothing I see that would block thing being merged in.

It's not really worth reviewing my others PRs until this is merged in, as they will all need some small changes before they're ready for a re-review

**Review by mrchrisadams — APPROVED** [review]

**Pull request merged**
